src/Collapse.php

```diff
@@ -4,195 +4,418 @@
 use Yiisoft\Html\Html;
+use Yiisoft\Html\Tag\Div;
+use Yiisoft\Widget\Widget;
 
-use function array_key_exists;
+use function implode;
 
 /**
- * Collapse renders an collapse bootstrap JavaScript component.
+ * Collapse renders a Bootstrap collapse component.
  *
- * For example
+ * For example:
  *
- * $collapse =Collapse::widget()
- *      ->withTitle('Foo')
- *      ->withContent('Bar');
+ * ```php
+ * echo Collapse::widget()
+ *     ->containerAttributes(['class' => 'row'])
+ *     ->items(
+ *         CollapseItem::to(
+ *             'Some placeholder content for the first collapse component of this multi-collapse example. ' .
+ *             'This panel is hidden by default but revealed when the user activates the relevant trigger.',
+ *             'multiCollapseExample1',
+ *             togglerContent: 'Toggle first element',
+ *             togglerAsLink: true,
+ *         ),
+ *         CollapseItem::to(
+ *             'Some placeholder content for the second collapse component of this multi-collapse example. ' .
+ *             'This panel is hidden by default but revealed when the user activates the relevant trigger.',
+ *             'multiCollapseExample2',
+ *             togglerContent: 'Toggle second element',
+ *         ),
+ *         CollapseItem::to(
+ *             togglerContent: 'Toggle both elements',
+ *             togglerMultiple: true,
+ *             ariaControls: 'multiCollapseExample1 multiCollapseExample2',
+ *       ),
+ *     )
+ *     ->render();
+ * ```
  *
- * echo $collapse->render();
- *
- * Or
- *
- * $collapse = $collapse->withToggle(false);
- *
- * echo '<p>' . $collapse->renderToggle() . '</p><div>Some other content</div>' . $collapse->render();
+ * @link https://getbootstrap.com/docs/5.3/components/collapse/
  */
-final class Collapse extends AbstractToggleWidget
+final class Collapse extends Widget
 {
-    private array $options = [];
-    private array $bodyOptions = [
-        'tag' => 'div',
-    ];
-    private string|Stringable $content = '';
-    private bool $horizontal = false;
-    private bool $collapsed = false;
-    private ?string $tagName = null;
-
-    public function getId(?string $suffix = '-collapse'): ?string
-    {
-        // TODO: fix the method call, there's no suffix anymore.
-        return $this->options['id'] ?? parent::getId($suffix);
-    }
+    private const NAME = 'collapse';
+    private const CARD = 'card';
+    private const CARD_BODY = 'card-body';
+    private const COLLAPSE_MULTIPLE = 'multi-collapse';
+    private array $attributes = [];
+    private array $cardBodyAttributes = [];
+    private bool $container = true;
+    private array $containerAttributes = [];
+    private array $cssClasses = [];
+    /** @var CollapseItem[] */
+    private array $items = [];
+    private string $togglerContainerTag = 'p';
+    private array $togglerContainerAttributes = [];
 
-    protected function toggleComponent(): string
+    /**
+     * Adds a sets of attributes.
```

Comment:
```suggestion
     * Adds a set of attributes.
```

Reply:
But the method adds an existing attribute to the general attributes, and it is the same phpdoc in `yiisoft/html`.